### PR TITLE
Corner cases for jcli tests

### PR DESCRIPTION
**Pull request by dkijania**
Signed-off-by: dkijania <dariusz.kijania@iohk.io>

### CODE REVIEW

File: tests/jcli/address/info.rs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+#[test]
+pub fn test_info_delegation_address() {
+    let private_key = jcli_wrapper::assert_key_generate("ed25519Extended");
+    let public_key = jcli_wrapper::assert_key_to_public_default(&private_key);
+
+    let private_key = jcli_wrapper::assert_key_generate("ed25519Extended");
+    let delegation_key = jcli_wrapper::assert_key_to_public_default(&private_key);
+    let account_address =
+        jcli_wrapper::assert_address_delegation_for_testing(&public_key, &delegation_key);
```
Comment (mzabaluev) [inline tests/jcli/address/info.rs]: Could have used some profile data structures to select between testing and production, rather than functions with duplicate code.

File: tests/jcli/address/single.rs [hunk 2]
```diff
@@ -85,9 +132,10 @@ pub fn test_delegation_address_made_of_random_string() {
 
     // Assertion changed due to issue #306. After fix please change it to correct one
     process_assert::assert_process_failed_and_contains_message(
-        jcli_wrapper::jcli_commands::get_address_delegation_command_default(
+        jcli_wrapper::jcli_commands::get_address_delegation_command(
             &public_key,
             &delegation_key,
+            true,
```
Comment (mzabaluev) [inline tests/jcli/address/single.rs]: The boolean parameter is not informative. Invent an enum with descriptive values?

File: tests/common/jcli_wrapper/mod.rs [hunk 3]
```diff
@@ -15,6 +15,12 @@ use super::process_utils::output_extensions::ProcessOutput;
 use std::collections::BTreeMap;
 use std::path::PathBuf;
 
+#[derive(PartialEq)]
+pub enum Discrimination {
```
Comment (mzabaluev) [inline tests/common/jcli_wrapper/mod.rs]: I'd call it `Profile` or maybe `BlockchainSelection` for clarity, but it's a little matter.

File: tests/common/jcli_wrapper/mod.rs [hunk 4]
```diff
@@ -15,6 +15,12 @@ use super::process_utils::output_extensions::ProcessOutput;
...
+    Production,
+    Test,
+}
+
```
Comment (mzabaluev) [inline tests/common/jcli_wrapper/mod.rs]: Tip: if you'd like to have a clearly designated default value for this, make `impl Default for Discrimination { ... }`, then you can use `Default::default()` as a constructor.

File: tests/common/jcli_wrapper/mod.rs (hunk 3 above)
Comment (mzabaluev) [reply tests/common/jcli_wrapper/mod.rs]: Unless "address discrimination" is an established term that I've missed.

File: tests/common/jcli_wrapper/mod.rs (hunk 4 above)
Comment (dkijania) [reply tests/common/jcli_wrapper/mod.rs]: Ok. will implement this in next PR
